src/dotnet/Telemetry.cs

```diff
@@ -63,7 +90,18 @@ public void TrackEvent(string eventName, IDictionary<string, string> properties,
             {
                 return;
             }
+            if(!_shouldLogTelemetry)
+            {
+                return;
+            }                
 
+            _trackEventTask = Task.Factory.StartNew(
```

Comment:
At this time there should only be the single trackEvent.   However, the only reason we even need to capture the Task to begin with is to add a little wait 100 milliseconds. So we only care about waiting for the latest sent task.  As the earlier ones should have completed by the time the last one finishes.